src/App.tsx

```diff
@@ -8,7 +8,100 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [user, setUser] = useState<User | 0>(0);
+  const [selectedId, setSelectedId] = useState(0);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | 0>(0);
+  const [todoSortByData, setTodoSortByData] = useState('');
+  const [searchQuery, setSearchQuery] = useState('');
+  const [isListLoading, setIsListLoading] = useState(false);
+  const [isModalLoading, setIsModalLoading] = useState(false);
+
+  const loadTodoList = async () => {
+    const loadedTodoList = await getTodos();
+
+    if (loadedTodoList) {
+      setIsListLoading(false);
+      setTodos(loadedTodoList);
+    }
+  };
+
+  const loadUser = async (todoId: number) => {
+    const loadedUsers = await getUser(todoId);
+
+    if (loadedUsers) {
+      setUser(loadedUsers);
+      setTimeout(() => {
+        setIsModalLoading(false);
+      }, 300);
```

Comment:
`setTimeout` is not a good approach. it would be better to update the state immediately
```suggestion
      setIsModalLoading(false);
```